src/mscorlib/src/System/Diagnostics/Stacktrace.cs

```diff
@@ -64,9 +64,6 @@ internal class StackFrameHelper : IDisposable
         private static Type s_symbolsType = null;
         private static MethodInfo s_symbolsMethodInfo = null;
 
-        [ThreadStatic]
-        private static int t_reentrancy = 0;
```

Comment:
It seems safer to leave the reentrancy check in place. Even if we do catch the exceptions, that won't prevent infinite recursion if the code within this block (and its complete call-graph) somewhere throws an exception and then tries to log the stack trace for it.